### PR TITLE
Faster CSV loading

### PR DESCRIPTION
**Pull request by mrzzzrm**
The is not the "fast" CSV parser you are looking for. But it reduces the loading time of a 600MB CSV into memory from 1,7s to 0,4s.
The actual parsing still takes ~20s for this file.

### CODE REVIEW

File: src/lib/import_export/csv_parser.cpp [hunk 1]
```diff
@@ -34,14 +34,25 @@ std::shared_ptr<Table> CsvParser::parse(const std::string& filename, const std::
     _meta = *csv_meta;
   }
 
+  _escaped_linebreak = std::string(1, _meta.config.delimiter_escape) + std::string(1, _meta.config.delimiter);
```
Comment (Bensk1) [inline src/lib/import_export/csv_parser.cpp]: Reading `delimiter` in the context of a line break made me wonder. The config has a `delimiter` which marks the end of a line and a `separator` that marks the end of a field. I believe in every other program, the entities indicating the end of a field are called `delimiters`. Should we rename that?

Comment (mrzzzrm) [reply src/lib/import_export/csv_parser.cpp]: Probably. That and a dozen other things in the CsvParser. Doesn't really make sense to tackle one (c/p) isolated thing in this PR.

Comment (Bensk1) [reply src/lib/import_export/csv_parser.cpp]: I guess you already maintain a private list. Do you want to publish it?

Comment (mrzzzrm) [reply src/lib/import_export/csv_parser.cpp]: No, I don't really do. We can have a quick chat about rewriting the damn thing at some point, if you want to.

Comment (Bensk1) [reply src/lib/import_export/csv_parser.cpp]: Sure.

File: src/lib/import_export/csv_parser.cpp [hunk 2]
```diff
@@ -198,14 +215,12 @@ size_t CsvParser::_parse_into_chunk(std::string_view csv_chunk, const std::vecto
 }
 
 void CsvParser::_sanitize_field(std::string& field) {
-  const std::string linebreak(1, _meta.config.delimiter);
-  const std::string escaped_linebreak =
-      std::string(1, _meta.config.delimiter_escape) + std::string(1, _meta.config.delimiter);
+  constexpr size_t ESCAPED_LINEBREAK_SIZE = 2;
```
Comment (mrks) [inline src/lib/import_export/csv_parser.cpp]: Mmmh, setting this as a constant makes it look like a developer-chosen constant, not like something that is dependent on `_escaped_linebreak.size()`. At least add a comment explaining its relation to `escaped_linebreak.size()` and why it always has to be 2.

Comment (mrzzzrm) [reply src/lib/import_export/csv_parser.cpp]: Using `escaped_linebreak.size()` again as it didn't make a difference anyway

**Pull request merged**